### PR TITLE
Move all contributors to a different file

### LINKED ISSUE #3847: [Bug]: Fix `Lychee Broken Link Checker` for Stack Overflow Links
### PyBaMM Version

Develop

### Python Version

3.11.5

### Describe the bug

PyBaMM's Lychee link checker is unable to get through Stack Overflow's security to verify links. 

### Steps to Reproduce

By using the `Check URLs using Lychee` GitHub Action of PyBaMM

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by kratman**
# Description

Moves the all contributors section out of the README.md file so that it can be ignored by Lychee.

Fixes #3847

## Type of change

Minor fix to make the CI more stable. Since the all contributors section contains links to contributors personal web profiles, they can cause lychee to fail if a users site is not working. Issues like this will become more common as we get more contributors.

- [x] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [x] No style issues: `$ pre-commit run` (or `$ nox -s pre-commit`) (see [CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [x] All tests pass: `$ python run-tests.py --all` (or `$ nox -s tests`)
- [x] The documentation builds: `$ python run-tests.py --doctest` (or `$ nox -s doctests`)

You can run integration tests, unit tests, and doctests together at once, using `$ python run-tests.py --quick` (or `$ nox -s quick`).

## Further checks:

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works

### CODE REVIEW

**Comment by kratman** [comment]
The link check fails because my new file is not on develop yet. So that will be fixed after this is merged

File: all_contributors.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    </tr>
+  </tbody>
+</table>
+
+<!-- markdownlint-restore -->
+<!-- prettier-ignore-end -->
+
+<!-- ALL-CONTRIBUTORS-LIST:END -->
+
+This project follows the [all-contributors](https://github.com/all-contributors/all-contributors) specification. Contributions of any kind welcome!
```
[the remark this answers is not in the document]
Comment (kratman) [reply all_contributors.md:115]: I thought that was some of the stuff that was generated which is why I moved it too. If it wasn't then I am fine with moving it there

**Comment by agriyakhetarpal** [comment]
Note: please exclude `all-contributors.md` from inclusion in the SDist by modifying `MANIFEST.in`

**Comment by kratman** [comment]
> Note: please exclude `all_contributors.md` from inclusion in the SDist by modifying `MANIFEST.in`

Done

**Pull request merged**